modules/diskq/qdisk.c

```diff
@@ -494,20 +494,21 @@ _try_to_load_queue(QDisk *self, GQueue *queue, LoadParameters *params)
      .type = #_type \
 }
 
+#define try_load_queue(queue) \
+( { \
+  gboolean retval = TRUE; \
+  LoadParameters queue ## _parameters = create_load_parameter_for(queue); \
+  if (!_try_to_load_queue(self, queue, &queue ## _parameters)) \
+    retval = FALSE; \
+  retval; \
+} )
+
 static gboolean
 _load_non_reliable_queues(QDisk *self, GQueue *qout, GQueue *qbacklog, GQueue *qoverflow)
 {
-  LoadParameters qout_parameters = create_load_parameter_for(qout);
-  if (!_try_to_load_queue(self, qout, &qout_parameters))
-    return FALSE;
```

Comment:
I think this repetition is not harmful at all, especially if you have to use macros and token pasting operators to replace them.

I understand the necessity of `create_load_parameter_for()`, but in my opinion the macro-based `try_load_queue()` adds unnecessary complexity to a straightforward/harmless repetition.